ethcore/src/migrations/upgrade_bloom.rs

```diff
@@ -0,0 +1,137 @@
+	let hash_count_entry = db.get(COL_ACCOUNT_BLOOM, ACCOUNT_BLOOM_HASHCOUNT_KEY)
+		.expect("Low-level database error");
+
+	hash_count_entry.is_some()
+}
+
+fn check_space_match(db: &Database) -> Result<(), usize> {
+	let db_space = db.get(COL_ACCOUNT_BLOOM, ACCOUNT_BLOOM_SPACE_KEY)
+		.expect("Low-level database error")
+		.map(|val| LittleEndian::read_u64(&val[..]) as usize)
+		// this was the initial size of the bloom which was not written in the database
+		.unwrap_or(1048576);
+
+	if db_space == ACCOUNT_BLOOM_SPACE { Ok(()) } else { Err(db_space) }
+}
+
+/// Account bloom upgrade routine. If bloom already present, does nothing.
+/// If database empty (no best block), does nothing.
+/// Can be called on upgraded database with no issues (will do nothing).
+pub fn upgrade_account_bloom(db_path: &Path) -> Result<(), Error> {
+	let mut progress = ::util::migration::Progress::default();
+
+	let path = try!(db_path.to_str().ok_or(Error::MigrationImpossible));
+	trace!(target: "migration", "Account bloom upgrade at {:?}", db_path);
+
+	let source = try!(Database::open(&DatabaseConfig {
+		max_open_files: 64,
+		cache_sizes: HashMap::new(),
+		compaction: CompactionProfile::default(),
+		columns: NUM_COLUMNS,
+		wal: true,
+	}, path));
+
+	let best_block_hash = match try!(source.get(COL_EXTRA, b"best")) {
+		// no migration needed
+		None => {
+			trace!(target: "migration", "No best block hash, skipping");
+			return Ok(());
+		},
+		Some(hash) => hash,
+	};
+	let best_block_header = match try!(source.get(COL_HEADERS, &best_block_hash)) {
+		// no best block, nothing to do
+		None => {
+			trace!(target: "migration", "No best block header, skipping");
+			return Ok(())
+		},
+		Some(x) => x,
+	};
+	let state_root = HeaderView::new(&best_block_header).state_root();
+
+	let db = ::std::sync::Arc::new(source);
+	let mut batch = DBTransaction::new(&db);
+
+	if check_bloom_exists(&*db) {
+		match check_space_match(&*db) {
+			Ok(_) => {
+				// bloom already exists and desired and stored spaces match, nothing to do
+				trace!(target: "migration", "Bloom already present of the right space, skipping");
+				return Ok(())
+			},
+			Err(wrong_size) => {
+				// nullify existing bloom entries
+				trace!(target: "migration", "Clearing old bloom of space {}", &wrong_size);
+				let mut key = [0u8; 8];
+				for idx in 0..(wrong_size as u64/8) {
+					LittleEndian::write_u64(&mut key, idx);
+					batch.put(COL_ACCOUNT_BLOOM, &key, &[0u8; 8]);
+
+					if idx % 10000 == 1 { progress.tick(); };
+				}
+
+				LittleEndian::write_u64(&mut key, ACCOUNT_BLOOM_SPACE as u64);
+				batch.put(COL_ACCOUNT_BLOOM, ACCOUNT_BLOOM_SPACE_KEY, &key);
+			},
+		}
+	}
+
+	println!("Adding/expanding accounts bloom (one-time upgrade)");
+	let bloom_journal = {
+		let mut bloom = Bloom::new(ACCOUNT_BLOOM_SPACE, DEFAULT_ACCOUNT_PRESET);
+		// no difference what algorithm is passed, since there will be no writes
+		let state_db = journaldb::new(
```

Comment:
It's also probably best to just take the pruning argument as a parameter -- we have it in the migration phase, so there's no point firing shots in the dark. I don't see why `refcounted` and `light` pruning wouldn't be supported with this migration.